pkg/appcommon/harness.go

```diff
@@ -32,9 +32,10 @@ var (
 )
 
 type Config struct {
-	InstrumentBuckets string `yaml:"instrument_buckets"`
-	EnableAuth        bool   `yaml:"enable_auth"`
-	ServiceName       string `yaml:"service_name"`
+	InstrumentBuckets string             `yaml:"instrument_buckets"`
+	EnableAuth        bool               `yaml:"enable_auth"`
+	ServiceName       string             `yaml:"service_name"`
+	Tracer            opentracing.Tracer `yaml:"-"`
```

Comment:
do we not want to set up a yaml value?  would be "tracer" I assume.

Reply:
I think "-" means it doesn't map to the yaml (the type isn't something we could express in yaml, unlike the other fields)

We're using the field programatically to pass an otel-bridge tracer:
```
	otel.SetTracerProvider(wrapperTracerProvider)
	opentracing.SetGlobalTracer(bridgeTracer)
+	appConfig.Tracer = bridgeTracer
```
( from https://github.com/grafana/app-o11y-kwl-endpoint/blob/9e5f1b9a19c98ecfa707d7b0cff8d6178488e879/cmd/kwl/main.go#L71 )

Reply:
ahhh I see. The appconfig is meant for config options that can be set by yaml, so if that can't be done here then the appconfig struct isn't the right place to do this.  Instead, Tracer should be a parameter on New(), like the prom Registerer. The logic below can remain mostly the same.

I only see two callsites for New() so it shouldn't be a big problem to update them both.

Reply:
Right - that makes sense! Thanks for pointing that out - New(...tracer Tracer) should do the trick 👍